src/main/java/core/AbstractGameState.java

```diff
@@ -234,16 +258,33 @@ void checkActionsInProgress() {
         }
     }
 
+    public final void endGame() {
```

Comment:
This is functionality, changing the game state, I think it's better suited on the FM. I'd also say making this not final, so that games can process the end of game differently if needed.

Reply:
Afterthought: I'm getting a bit concerned of functionality split between FM and turn order. We have setup and endGame on FM, but the turn order has functionality for start round (which is often also done in setup) and end round, also calling end game in this format from the game state on max rounds reached. I think we need to restructure this a bit so it's more clear who does what...

Reply:
"I think we need to restructure this a bit so it's more clear who does what..."

Yes - let's have that discussion. In this particular case putting endGame() on the GameState makes sense I think purely because the corresponding FM.endGame() is only used in one place (in Game); while code on the GameState to end the game is called directly all over the place. This is an issue in MCTS, RHEA. RMHC etc where currently the game is ended using GameState methods.

Very happy to move this to use an FM method - but it requires rather more refactoring.